### PR TITLE
[Move] Implement Powder (with edge case)

### PR DESCRIPTION
**Pull request by innerthunder**
## What are the changes the user will see?

This partially implements the move Powder. In this implementation, Powder works as expected for the most part, except that Fire-type moves generated by Dancer incorrectly ignore the effect.

## Why am I making these changes?

Powder is a fun move.

## What are the changes from a developer perspective?

- `data/battler-tags`: New `PowderTag` implements the effects of Powder, including the move-canceling and self-damage effect when the afflicted Pokemon uses a Fire-type move.
- `data/move`: Powder now uses an `AddBattlerTagAttr` to apply the new `PowderTag` onto the target for a turn.
- `test/moves/powder.test`: New integration test suite for Powder's effects.

### Screenshots/Videos

Using Powder against an enemy (the enemy is forced to use Ember with overrides)

https://github.com/user-attachments/assets/4fe87126-119d-4457-8326-696f33e76ec5

## How to test the changes?

`npm run test powder` (though this also runs other unrelated tests... `powder.test.ts` is the important one!)

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I add placeholders for them in locales?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by frutescens** [comment]
Would it be possible for the move Tar Shot to have a similar implementation? https://bulbapedia.bulbagarden.net/wiki/Tar_Shot_(move)

**Comment by Adri1** [comment]
In the battler-tags.json texts is possible to have the possibilty to mention the move Powder interacts with?
Because eveen tho English text don't mention it, it is mentioned in ALL other languages

Examples :
 - English: When the flame touched the powder on the Pokémon, it exploded! (nothing)
 - French: La nuée de poudre entre en réaction avec [MOVE NAME INTECATING WITH POWDER] et explose !"
 - Japanese: [MOVE NAME INTECATING WITH POWDER]に　反応して　ふんじんが　爆発した！

File: src/test/moves/powder.test.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+import { Abilities } from "#enums/abilities";
+import { Moves } from "#enums/moves";
+import { Species } from "#enums/species";
+import { BerryPhase } from "#app/phases/berry-phase";
+import { MoveResult } from "#app/field/pokemon";
+import { Type } from "#enums/type";
+import { MoveEffectPhase } from "#app/phases/move-effect-phase";
+import { StatusEffect } from "#enums/status-effect";
+
+const TIMEOUT = 20 * 1000;
```
Comment (DayKev) [inline src/test/moves/powder.test.ts:13]:
```suggestion
```
The default timeout was set to 20s across all tests, so we don't need to specify it manually anymore.

**Review by DayKev — DISMISSED** [review]

File: src/test/moves/powder.test.ts [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override.battleType("single");
+
+    game.override
+      .enemySpecies(Species.SNORLAX)
+      .enemyLevel(100)
+      .enemyMoveset(Array(4).fill(Moves.EMBER))
```
Comment (torranx) [inline src/test/moves/powder.test.ts:34]: can't look it up now but didn't we introduce something that removes the need to fill the array with a singular move?

**Review by DayKev — APPROVED** [review]

**Review by frutescens — APPROVED** [review]

**Pull request merged**